### PR TITLE
PathKit: port to Windows

### PR DESCRIPTION
**Pull request by compnerd**
Windows does not provide the Unix `glob` and `fnmatch` functions. Emulate these with the closes equivalences (`FindFirstFileW` and `PathMatchSpecExW`).  This should allow us to use this library on Windows.

This does not cover the tests which will need to be adjusted subsequently.

### CODE REVIEW

File: Sources/PathKit.swift [hunk 1]
```diff
@@ -587,6 +589,26 @@ extension Path {
...
   public static func glob(_ pattern: String) -> [Path] {
+#if os(Windows)
+    var fdData: WIN32_FIND_DATAW = WIN32_FIND_DATAW()
+    let hSearch: HANDLE = pattern.withCString(encodedAs: UTF16.self) {
+      FindFirstFileW($0, &fdData)
+    }
+    if (hSearch == INVALID_HANDLE_VALUE) {
+      return []
+    }
+    defer { FindClose(hSearch) }
```
Comment (djbe) [inline Sources/PathKit.swift:600]: Shouldn't this defer be above that first `if`?

Edit: or is it because you can't close an invalid handle 🤔

**Review by djbe — CHANGES_REQUESTED** [review]
In https://github.com/stencilproject/Stencil/pull/343 you use `withUnsafeFileSystemRepresentation` to normalise the path (expand `~`, etc…). I'm assuming that change will also be needed here? To replace the `NSString` dependency?

File: Sources/PathKit.swift (hunk 1 above)
Comment (compnerd) [reply Sources/PathKit.swift:600]: The edit is correct - it would close an invalid handle.

**Comment by compnerd** [comment]
Turns out that this is woefully insufficient and the changes required to support this is too invasive.  I'm going to close this for now, feel free to use it as a base if you feel motivated to port the library.

**Pull request closed without merging**